### PR TITLE
[BUG] Fix source param calculation

### PR DESCRIPTION
**Pull request by nitbharambe**
Might affect https://github.com/PowerGridModel/power-grid-model/pull/666

### CODE REVIEW

**Review by mgovers — COMMENTED** [review]
Oh wow this could indeed explain the issues @figueroa1395 is running into on #666

**Comment by TonyXiang8787** [comment]
Are you sure about this?

I think we use the `a` matrix in the order of 1, 2, 0

**Comment by nitbharambe** [comment]
@TonyXiang8787 We have our transformation matrices in this way:
```
// symmetric component matrix
inline ComplexTensor<asymmetric_t> get_sym_matrix() {
    ComplexTensor<asymmetric_t> m;
    m << 1.0, 1.0, 1.0, 1.0, a2, a, 1.0, a, a2;
    return m;
}
inline ComplexTensor<asymmetric_t> get_sym_matrix_inv() {
    ComplexTensor<asymmetric_t> m;
    m << 1.0, 1.0, 1.0, 1.0, a, a2, 1.0, a2, a;
    m = m / 3.0;
    return m;
}
```
The order does correspond to 0,1,2. At most places we have initialized using y1, y0 directly.

Regarding validation cases, we have mostly tried to create the ideal slack bus. (High `sk`). Hence, all the `z01_ratio` used are `1.0` as to not introduce any unnecessary asymmetric-ness to make the electrical modelling not on-to-one. This would let the bug not surface up.

We should indeed add validation case for a higher `z01_ratio`.

**Comment by TonyXiang8787** [comment]
Please then first add a validation case to reproduce the error.

And then push the fix.

File: tests/cpp_unit_tests/test_source.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ TEST_CASE("Test source") {
     ComplexTensor<asymmetric_t> const sym_matrix = get_sym_matrix();
     ComplexTensor<asymmetric_t> const sym_matrix_inv = get_sym_matrix_inv();
     ComplexTensor<asymmetric_t> y012;
-    y012 << y1, 0.0, 0.0, 0.0, y1, 0.0, 0.0, 0.0, y0;
+    y012 << y0, 0.0, 0.0, 0.0, y1, 0.0, 0.0, 0.0, y1;
```
Comment (Jerry-Jinfeng-Guo) [inline tests/cpp_unit_tests/test_source.cpp:45]: @figueroa1395 also touched on this, perhaps it's good idea to isolate issues.

File: power_grid_model_c/power_grid_model/include/power_grid_model/component/source.hpp [hunk 2]
```diff
@@ -48,12 +48,7 @@ class Source : public Appliance {
         if constexpr (is_symmetric_v<sym>) {
             return y1_ref_;
         } else {
-            ComplexTensor<asymmetric_t> const sym_matrix = get_sym_matrix();
-            ComplexTensor<asymmetric_t> const sym_matrix_inv = get_sym_matrix_inv();
-            ComplexTensor<asymmetric_t> y012;
-            y012 << y1_ref_, 0.0, 0.0, 0.0, y1_ref_, 0.0, 0.0, 0.0, y0_ref_;
-            ComplexTensor<asymmetric_t> yabc = dot(sym_matrix, y012, sym_matrix_inv);
-            return yabc;
+            return ComplexTensor<asymmetric_t>{(2.0 * y1_ref_ + y0_ref_) / 3.0, (y0_ref_ - y1_ref_) / 3.0};
```
[the remark this answers is not in the document]
Comment (nitbharambe) [reply power_grid_model_c/power_grid_model/include/power_grid_model/component/source.hpp:51]: Yes, this is what the operation simplifies to. We have similar thing being done in `shunt` btw.

**Review by Jerry-Jinfeng-Guo — APPROVED** [review]
Looks good to me

**Review by TonyXiang8787 — APPROVED** [review]

**Pull request merged**